### PR TITLE
Detect submissions to the same exercise with a different group

### PR DESCRIPTION
**Pull request by Taikelenn**
# Description of the PR
Whenever a student submits an assignment, the plugin stores the information about the group which the user has selected. Then, if the user attempts to submit again with a different group, a warning message will appear in the submission dialog.

To achieve this, the Group class now contains the user IDs. The plugin can construct a "group memberwise ID" such that two groups with the same members have the same memberwise ID.

obraz

### CODE REVIEW

**Review by OlliKiljunen — COMMENTED** [review]
Nice work! :+1: I made some comments I'd like to discuss before approval.

File: src/main/java/fi/aalto/cs/apluscourses/intellij/actions/SubmitExerciseAction.java
Comment (Taikelenn) [reply src/main/java/fi/aalto/cs/apluscourses/intellij/actions/SubmitExerciseAction.java:341]: Fixed

File: src/main/java/fi/aalto/cs/apluscourses/intellij/actions/SubmitExerciseAction.java
Comment (Taikelenn) [reply src/main/java/fi/aalto/cs/apluscourses/intellij/actions/SubmitExerciseAction.java:315]: Fixed

File: src/main/java/fi/aalto/cs/apluscourses/intellij/utils/Interfaces.java
Comment (Taikelenn) [reply src/main/java/fi/aalto/cs/apluscourses/intellij/utils/Interfaces.java:308]: It would be nice, I chose the current solution because it was objectively easier to implement. We could go for the way you proposed, but in that case I'd have that as a separate issue and handle both this & duplicate submission checker there

File: src/main/java/fi/aalto/cs/apluscourses/model/Group.java
Comment (Taikelenn) [reply src/main/java/fi/aalto/cs/apluscourses/model/Group.java:15]: Which class do you mean? `fi.aalto.cs.apluscourses.model.Student` doesn't work because it also requires a student ID, neither does `fi.aalto.cs.apluscourses.model.User` because that's something for authentication.

**Review by OlliKiljunen — APPROVED** [review]
Oh, I had already re-reviewed this earlier but forgot to press submit button. 🤦‍♂️ Anyway, have an approval.

**Pull request merged**
